counsel.el

```diff
@@ -5138,6 +5141,37 @@ The optional `BUILD-ROOT' is useful for other helpers that have found
      local-files)
     compiles))
 
+(defun counsel--find-build-subdir (root)
+  "Return builds sub-directory of `ROOT', if one exists."
+  (cl-some
+   #'(lambda (x)
+       (let ((check (format "%s%s" root x)))
+         (when (file-directory-p check)
+           check)))
+   (list "build" "builds" "bld")))
```

Comment:
Has to be hookable.  I personally prefer `.build` next to `.git`.

Reply:
I could just make it a defvar counsel-compile-potential-build-dirs and use that?

Reply:
I think it should be `defcustom` and I would just call it `counsel-compile-build-directories` as `-directories` suffix is consistent with both vanilla Emacs and some existing `counsel` `defcustom`s, while `potential` isn't strictly necessary here.